src/filters/ws.rs

```diff
@@ -284,19 +206,26 @@ impl Stream for WebSocket {
     type Item = Result<Message, crate::Error>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
-        match (*self).with_context(Some(cx), |s| s.read_message()) {
-            Ok(item) => Poll::Ready(Some(Ok(Message { inner: item }))),
-            Err(::tungstenite::Error::Io(ref err)) if err.kind() == io::ErrorKind::WouldBlock => {
+        match (*self).with_context(Some(cx), |s, cx| s.poll_next(cx.unwrap())) {
+            Poll::Ready(Some(Ok(item))) => Poll::Ready(Some(Ok(Message { inner: item }))),
+            Poll::Ready(Some(Err(tungstenite::Error::Io(ref err))))
+                if err.kind() == io::ErrorKind::WouldBlock =>
```

Comment:
same as below

Reply:
only this one's left

Reply:
this was also removed in the last commit